### ISSUE TITLE
Cancelled upload files aren't immediately deleted

### ISSUE THREAD

**Issue by zipiju** [storj/storj#7058]
Noticed this on the forum, where an user was asked to create a bug report after finding this bug, but I'm seeing none here, so creating one after was able to reproduce the issue.

It looks like the node isn't actually deleting files from uploads that were cancelled at some point.
This probably is a significant source of garbage and additional IOPS, especially in situations where the upload success rate tanks - either because the disk subsystem cannot keep up (as during the filewalks, and even with SSD cache in front of the platter drive), or when the circuit bandwidth is all used up.

As per the logs, these example uploads were all cancelled, the files were however still created in the blobs folder:
```
# tail -f /var/log/storj/storj.log | grep cancel
2024-07-19T01:35:30+02:00       INFO    piecestore      upload canceled (race lost or node shutdown)    {"Process": "storagenode", "Piece ID": "B4PJDU3TYYRUA2VJTRAPJ65T2Z3R6PZ76GVCJNCI6N4WU3B7MMKA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "109.61.92.82:39516"}
2024-07-19T01:35:30+02:00       INFO    piecestore      upload canceled (race lost or node shutdown)    {"Process": "storagenode", "Piece ID": "UZ7YJRS3JNNBOK22N562O7SHNNXWLFJOF7IYVVGRACZTJONLG7TQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "79.127.213.34:38270"}
2024-07-19T01:35:30+02:00       INFO    piecestore      upload canceled (race lost or node shutdown)    {"Process": "storagenode", "Piece ID": "EBSPEU6RN5765OKD3I2EBRW73FROL3MEKCHTSQQTIK2KV3YFSCUQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "79.127.213.33:33718"}
2024-07-19T01:35:31+02:00       INFO    piecestore      upload canceled (race lost or node shutdown)    {"Process": "storagenode", "Piece ID": "RTPUVF3NFLXQU6E2WB4TMH5XYEPAPHKBPYQJBYQVF7YO2ENR2KSA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "79.127.205.228:50688"}

# ls -lah /mnt/storj-data/data/storage/blobs/ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa/b4/pjdu3tyyrua2vjtrapj65t2z3r6pz76gvcjnci6n4wu3b7mmka.sj1 
-rw-r--r-- 1 storj storj 1.3K Jul 19 01:35 /mnt/storj-data/data/storage/blobs/ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa/b4/pjdu3tyyrua2vjtrapj65t2z3r6pz76gvcjnci6n4wu3b7mmka.sj1
# ls -lah /mnt/storj-data/data/storage/blobs/ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa/uz/7yjrs3jnnbok22n562o7shnnxwlfjof7iyvvgracztjonlg7tq.sj1 
-rw-r--r-- 1 storj storj 16K Jul 19 01:35 /mnt/storj-data/data/storage/blobs/ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa/uz/7yjrs3jnnbok22n562o7shnnxwlfjof7iyvvgracztjonlg7tq.sj1
# ls -lah /mnt/storj-data/data/storage/blobs/ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa/eb/speu6rn5765okd3i2ebrw73frol3mekchtsqqtik2kv3yfscuq.sj1 
-rw-r--r-- 1 storj storj 15K Jul 19 01:35 /mnt/storj-data/data/storage/blobs/ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa/eb/speu6rn5765okd3i2ebrw73frol3mekchtsqqtik2kv3yfscuq.sj1
# ls -lah /mnt/storj-data/data/storage/blobs/ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa/rt/puvf3nflxqu6e2wb4tmh5xyepaphkbpyqjbyqvf7yo2enr2ksa.sj1 
-rw-r--r-- 1 storj storj 30K Jul 19 01:35 /mnt/storj-data/data/storage/blobs/ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa/rt/puvf3nflxqu6e2wb4tmh5xyepaphkbpyqjbyqvf7yo2enr2ksa.sj1
```

I would say this should be reworked to hold the piece all in memory until the upload is confirmed or cancelled, and only after this decision is made it should be written to the drive.
There is this filestore.write-buffer-size configuration option (set to 4MiB on this particular node), which would suggest this would be the case, but this apparently isn't working correctly.

**Comment by storjrobot** [comment]
This issue has been mentioned on **Storj Community Forum (official)**. There might be relevant details there:

https://forum.storj.io/t/two-weeks-working-for-free-in-the-waste-storage-business/26854/186

**Comment by MaxMaximus** [comment]
I was the user who initially reported this issue on the forum. But at that moment there was no time to issue a formal bug report here either. Now i have some free time, but since it has already been created, I will add my data here so as not to create duplicates of issues.

I can confirm issue on my nodes running on windows as native service. Including latest manually updated version of stragenode (v. 1.108.3).
```
2024-07-18T10:56:29+03:00	INFO	piecestore	upload started	{"Piece ID": "JQDWSGC4L5LCCIXXOKF5BQX37W7EST3T6O4PRQAQ6ZWIFSCXZDMA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT_REPAIR", "Remote Address": "5.161.243.123:34658", "Available Space": 1967392485923}
---cut---
2024-07-18T10:56:33+03:00	INFO	piecestore	upload canceled (race lost or node shutdown)	{"Piece ID": "JQDWSGC4L5LCCIXXOKF5BQX37W7EST3T6O4PRQAQ6ZWIFSCXZDMA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT_REPAIR", "Remote Address": "5.161.243.123:34658"}
```
! note: i also search log by piece id (to make sure piece is NOT uploaded later again without cancellation) and this is only strings found for this piece id

Let’s go to /blobs/ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa/jq/
where “ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa” is a storage dir for US1 satellites with id “12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S”
and /jq/ is sub-dir(prefix) taken as first two chars from piece ID.
And guess what we found there?

```
dir D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\jq\dwsgc4l5lccixxokf5bqx37w7est3t6o4prqaq6zwifscxzdma.sj1
 Folder content D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\jq

18.07.2024  10:56         2 319 872 dwsgc4l5lccixxokf5bqx37w7est3t6o4prqaq6zwifscxzdma.sj1
             1  file      2 319 872 bytes
```
Fresh portion of 2.3 MB of unaccounted garbage waiting next few weeks for next BF+GC+TFW runs!
I checked it few times more - its not a single glitch but usual (can not call it “normal”) behavior of current storagenode. 
Another one:
```
2024-07-19T05:44:44+03:00	INFO	piecestore	upload started	{"Piece ID": "TPJAO4BMHA7GKDZZ5KQFJ4BPZ4GEGODW57IBPL7V75AUGTZTH5ZQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "79.127.205.228:44080", "Available Space": 1919843000923}
2024-07-19T05:44:44+03:00	INFO	piecestore	upload canceled (race lost or node shutdown)	{"Piece ID": "TPJAO4BMHA7GKDZZ5KQFJ4BPZ4GEGODW57IBPL7V75AUGTZTH5ZQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "79.127.205.228:44080"}
```
```
dir D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tp\JAO4BMHA7GKDZZ5KQFJ4BPZ4GEGODW57IBPL7V75AUGTZTH5ZQ.*
 Folder content D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tp

19.07.2024  05:44         37 376 jao4bmha7gkdzz5kqfj4bpz4gegodw57ibpl7v75augtzth5zq.sj1
             1  file      37 376 bytes
```
I also checked the failed uploads in the same way. But it looks like in this case the files are not saved to /blobs/, i see this issue only for canceled uploads.

**Comment by mniewrzal** [comment]
I looked at this issue and as you noticed there is a case where piece will stay on disk when upload was canceled (log entry `upload canceled (race lost or node shutdown)`). I'm hesitant to fix it just be removing this piece during upload because it will generate more cpu and disk usage and usually node is loosing upload because some sort of overload (or just high cpu/disk usage). I think better approach would be to delay piece creation/write until write buffer is full. It won't cause that all canceled pieces won't be saved to disk but should reduce that number significantly, especially for systems with high `filestore.write-buffer-size`. This require a bit more changes so I cannot say exactly when we will try to do this.

**Comment by MaxMaximus** [comment]
Yes, of course, it will be much better just not to start writing data to a file on disk in a first place (not even create/open it) while the data from the partial upload fits in the write buffer (does not exceed the RAM limit set in the `filestore.write-buffer-size` parameter)
And in case of upload cancellation or upload failure, simply discard now useless partially uploaded data in RAM without creating (even temporarily) any files on disk at all.

But I saw a mention on the Storj forum from another user that there is already an immediate deletion of files created for partial uploads in the code of the storagenode. Here in this message:  https://forum.storj.io/t/two-weeks-working-for-free-in-the-waste-storage-business/26854/181
And my thought (guess) for what reason the file can still remain on the disk despite the presence of the code that should delete it (in next message from the same link above).

I don't know if this data is correct or not (because I haven't personally studied the code), but it's worth checking - maybe this functionality is still in the code now, but for some reason it doesn't work properly.
And it just needs to be fixed.

And in any case, please note that the processing of **canceled** and **failed** uploads now proceeds in different ways - in the 2nd case, the files on the disk do NOT remain unlike the first one. I've checked this personally and I'm sure of it.
Although, in theory, the logic of processing all partial unfinished uploads should be the same - the reason why the upload was not successfully completed is not important for the operation of the write buffer and the piece store module.

I do not know what happens in the case of failed uploads - whether the files are not written initially (while they are placed in the buffer), or they are written, but deleted. But the handling of cancelled and failed downloads is definitely different. 

P.S.
I can check the latter myself a little later if I find some more free time - by looking at disk activity, you can determine whether files are being created (and then immediately deleted) for failed uploads or they are simply not created at all.

**Comment by MaxMaximus** [comment]
Checked (node v 1.108.3 as Windows service, `filestore.write-buffer-size: 2.5 MiB`)
For failed uploads there are continues cycles of ( save partial upload to disk --> immediately delete created file):
storagenode.log
```
2024-07-27T06:54:08+03:00	INFO	piecestore	upload started	{"Piece ID": "C22W57USTI6OJO3TEIFXQ74G3EUSV5TDKL7LGQMJ5ATYGKWFAHYA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "79.127.201.210:37108", "Available Space": 1862005894198}
---cut---
2024-07-27T06:54:11+03:00	ERROR	piecestore	upload internal error	{"error": "manager closed: unexpected EOF", "errorVerbose": "manager closed: unexpected EOF\n\tgithub.com/jtolio/noiseconn.(*Conn).readMsg:225\n\tgithub.com/jtolio/noiseconn.(*Conn).Read:171\n\tstorj.io/drpc/drpcwire.(*Reader).read:68\n\tstorj.io/drpc/drpcwire.(*Reader).ReadPacketUsing:113\n\tstorj.io/drpc/drpcmanager.(*Manager).manageReader:229"}
2024-07-27T06:54:11+03:00	ERROR	piecestore	upload failed	{"Piece ID": "C22W57USTI6OJO3TEIFXQ74G3EUSV5TDKL7LGQMJ5ATYGKWFAHYA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "79.127.201.210:37108", "Size": 262144, "error": "manager closed: unexpected EOF", "errorVerbose": "manager closed: unexpected EOF\n\tgithub.com/jtolio/noiseconn.(*Conn).readMsg:225\n\tgithub.com/jtolio/noiseconn.(*Conn).Read:171\n\tstorj.io/drpc/drpcwire.(*Reader).read:68\n\tstorj.io/drpc/drpcwire.(*Reader).ReadPacketUsing:113\n\tstorj.io/drpc/drpcmanager.(*Manager).manageReader:229"}
```
disk i/o:
```
"Time of Day","Process Name","PID","Operation","Path","Result","Detail"
"6:54:08,6553661","storagenode.exe","3560","CreateFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\c2\2w57usti6ojo3teifxq74g3eusv5tdkl7lgqmj5atygkwfahya.sj1","SUCCESS","Desired Access: Generic Read/Write, Disposition: OverwriteIf, Options: Synchronous IO Non-Alert, Non-Directory File, Disallow Exclusive, Attributes: N, ShareMode: Read, Write, AllocationSize: 0, OpenResult: Created"
"6:54:11,6123994","storagenode.exe","3560","CloseFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\c2\2w57usti6ojo3teifxq74g3eusv5tdkl7lgqmj5atygkwfahya.sj1","SUCCESS",""
"6:54:11,6125418","storagenode.exe","3560","CreateFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\c2\2w57usti6ojo3teifxq74g3eusv5tdkl7lgqmj5atygkwfahya.sj1","SUCCESS","Desired Access: Read Attributes, Delete, Disposition: Open, Options: Non-Directory File, Open Reparse Point, Attributes: n/a, ShareMode: Read, Write, Delete, AllocationSize: n/a, OpenResult: Opened"
"6:54:11,6125830","storagenode.exe","3560","QueryAttributeTagFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\c2\2w57usti6ojo3teifxq74g3eusv5tdkl7lgqmj5atygkwfahya.sj1","SUCCESS","Attributes: ANCI, ReparseTag: 0x0"
"6:54:11,6126312","storagenode.exe","3560","SetDispositionInformationFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\c2\2w57usti6ojo3teifxq74g3eusv5tdkl7lgqmj5atygkwfahya.sj1","SUCCESS","Delete: True"
"6:54:11,6126827","storagenode.exe","3560","CloseFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\c2\2w57usti6ojo3teifxq74g3eusv5tdkl7lgqmj5atygkwfahya.sj1","SUCCESS",""
```
And for canceled upload its create+write file without deletion:
```
2024-07-27T07:23:22+03:00	INFO	piecestore	upload started	{"Piece ID": "TJXDEMN2QVFOBWUJWC3TRMRZ3JRDRLRHHMXU5S72PV47D5C3FZAQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "46.20.115.202:38608", "Available Space": 1861235533878}
---cut---
2024-07-27T07:23:27+03:00	INFO	piecestore	upload canceled (race lost or node shutdown)	{"Piece ID": "TJXDEMN2QVFOBWUJWC3TRMRZ3JRDRLRHHMXU5S72PV47D5C3FZAQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Remote Address": "46.20.115.202:38608"}
```
```
"Time of Day","Process Name","PID","Operation","Path","Result","Detail","Duration"
"7:23:22,1936512","storagenode.exe","3560","CreateFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","Desired Access: Generic Read/Write, Disposition: OverwriteIf, Options: Synchronous IO Non-Alert, Non-Directory File, Disallow Exclusive, Attributes: N, ShareMode: Read, Write, AllocationSize: 0, OpenResult: Created","0.0001354"
"7:23:27,0865688","storagenode.exe","3560","WriteFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","Offset: 512, Length: 2 319 360, Priority: Normal","0.0019166"
"7:23:27,0867319","storagenode.exe","3560","ReadFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","END OF FILE","Offset: 0, Length: 32 768, I/O Flags: Non-cached, Paging I/O, Synchronous Paging I/O, Priority: Normal","0.0000057"
"7:23:27,0885643","storagenode.exe","3560","WriteFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","Offset: 0, Length: 512","0.0000166"
"7:23:27,0886585","storagenode.exe","3560","CloseFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000343"
"7:23:27,0887130","storagenode.exe","3560","IRP_MJ_CLOSE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000078"
"7:23:27,4168599","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000039"
"7:23:27,4168731","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000013"
"7:23:27,4468720","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000036"
"7:23:27,4468852","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:27,8368737","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000030"
"7:23:27,8368857","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:27,8668876","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000036"
"7:23:27,8669005","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:28,3168906","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000036"
"7:23:28,3169032","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:28,3469024","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000030"
"7:23:28,3469144","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:28,8269130","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000036"
"7:23:28,8269259","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:28,8569212","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000030"
"7:23:28,8569347","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:29,2169348","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000034"
"7:23:29,2169475","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:29,2469361","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000033"
"7:23:29,2469485","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:29,5769311","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000027"
"7:23:29,5769485","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000015"
"7:23:29,6069528","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000033"
"7:23:29,6069660","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000015"
"7:23:29,9369550","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000033"
"7:23:29,9369673","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:29,9669695","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000051"
"7:23:29,9669866","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000015"
"7:23:30,2969641","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000033"
"7:23:30,2969767","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:30,3570058","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000060"
"7:23:30,3570220","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000021"
"7:23:30,7169788","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000033"
"7:23:30,7169902","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:30,7769931","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000033"
"7:23:30,7770051","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:31,1188094","System","4","WriteFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","Offset: 0, Length: 1 048 576, I/O Flags: Non-cached, Paging I/O, Synchronous Paging I/O, Priority: Normal","0.0906488"
"7:23:31,1970062","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000040"
"7:23:31,1970192","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:31,2095966","System","4","SetEndOfFileInformationFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndOfFile: 1 048 576","0.0000184"
"7:23:31,2096279","System","4","CreateFileMapping","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","SyncType: SyncTypeOther","0.0000015"
"7:23:31,2096381","System","4","FASTIO_RELEASE_FOR_SECTION_SYNCHRONIZATION","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:31,4670116","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000036"
"7:23:31,4670248","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000015"
"7:23:31,7370260","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000039"
"7:23:31,7370392","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:32,0070307","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000040"
"7:23:32,0070437","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000015"
"7:23:32,2770418","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000030"
"7:23:32,2770557","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:32,5770508","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000034"
"7:23:32,5770635","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000009"
"7:23:32,8770614","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000039"
"7:23:32,8770752","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:33,1125510","System","4","WriteFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","Offset: 1 048 576, Length: 1 048 576, I/O Flags: Non-cached, Paging I/O, Synchronous Paging I/O, Priority: Normal","0.0022701"
"7:23:33,1149493","System","4","SetEndOfFileInformationFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndOfFile: 2 097 152","0.0000192"
"7:23:33,1149842","System","4","CreateFileMapping","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","SyncType: SyncTypeOther","0.0000012"
"7:23:33,1149932","System","4","FASTIO_RELEASE_FOR_SECTION_SYNCHRONIZATION","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:33,2070292","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000030"
"7:23:33,2070449","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000018"
"7:23:33,5670420","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000030"
"7:23:33,5670546","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:33,9270584","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000039"
"7:23:33,9270710","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:34,3470688","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000040"
"7:23:34,3470839","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:34,7970830","System","4","FASTIO_ACQUIRE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndingOffset: 18 446 735 827 434 588 840","0.0000030"
"7:23:34,7970944","System","4","FASTIO_RELEASE_FOR_MOD_WRITE","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
"7:23:35,1112141","System","4","WriteFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","Offset: 2 097 152, Length: 225 280, I/O Flags: Non-cached, Paging I/O, Synchronous Paging I/O, Priority: Normal","0.0007961"
"7:23:35,1120617","System","4","SetEndOfFileInformationFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","EndOfFile: 2 319 872","0.0000174"
"7:23:35,1120936","System","4","CreateFileMapping","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","SyncType: SyncTypeOther","0.0000009"
"7:23:35,1121125","System","4","FASTIO_RELEASE_FOR_SECTION_SYNCHRONIZATION","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","","0.0000012"
```

**Comment by MaxMaximus** [comment]
I also noticed an interesting thing that applies not only to canceled uploads, but also to regular (successful) ones. 
Initially, the file is written with an offset of 512 bytes (not from the very beginning of the file).
After writing a file with a piece, an attempt to read this file immediately follows, only after that the first 512 bytes of the file are written:
```
"7:23:27,0865688","storagenode.exe","3560","WriteFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","Offset: 512, Length: 2 319 360, Priority: Normal","0.0019166"
"7:23:27,0867319","storagenode.exe","3560","ReadFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","END OF FILE","Offset: 0, Length: 32 768, I/O Flags: Non-cached, Paging I/O, Synchronous Paging I/O, Priority: Normal","0.0000057"
"7:23:27,0885643","storagenode.exe","3560","WriteFile","D:\Storj_Data\blobs\ukfu6bhbboxilvt7jrwlqk7y2tapb5d2r2tsmj2sjxvw5qaaaaaa\tj\xdemn2qvfobwujwc3trmrz3jrdrlrhhmxu5s72pv47d5c3fzaq.sj1","SUCCESS","Offset: 0, Length: 512","0.0000166"
```
What is it for?
I can assume that once before this was done to check the correctness of writing to the file, after which the headers /metadata are recorded stored in the first 512 bytes of the file.
But now with the f-sync disabled (which is now the default setting), such an attempt to read a newly written file with the "do not use cache" setting (I/O Flags: Non-cached) leads to the "END OF FILE" error. Because at this point the file (on a physical disk) is actually empty, all the data is in the cache and their actual writing to the disk begins a little later. In this example, after ~0.35 seconds delay when system start actual writing to disk - see the full log in the previous post.

**Comment by storjrobot** [comment]
This issue has been mentioned on **Storj Community Forum (official)**. There might be relevant details there:

https://forum.storj.io/t/when-will-uncollected-garbage-be-deleted/27327/52

**Comment by storjrobot** [comment]
This issue has been mentioned on **Storj Community Forum (official)**. There might be relevant details there:

https://forum.storj.io/t/when-will-uncollected-garbage-be-deleted/27327/58

**Comment by storjrobot** [comment]
This issue has been mentioned on **Storj Community Forum (official)**. There might be relevant details there:

https://forum.storj.io/t/when-will-uncollected-garbage-be-deleted/27327/113

**Comment by storjrobot** [comment]
This issue has been mentioned on **Storj Community Forum (official)**. There might be relevant details there:

https://forum.storj.io/t/drop-bloom-filters-bring-back-direct-deletes/27457/21
